File: app/src/main/java/io/github/maboueita/architecturecomponentssample/BestSellersListFragment.java
package io.github.maboueita.architecturecomponentssample;


import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import io.github.maboueita.architecturecomponentssample.data.OverviewApiResponse;

public class BestSellersListFragment extends LifecycleFragment {

    private static final String TAG = BestSellersListFragment.class.getSimpleName();

    private OverviewApiResponse overviewApiResponse;

    public BestSellersListFragment() {
        // Required empty public constructor
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        overviewApiResponse = ViewModelProviders.of(this).get(OverviewApiResponse.class);
        overviewApiResponse.init();

        if (overviewApiResponse.getResults() != null) {
            overviewApiResponse.getResults().observe(this, results -> {
                Log.d(TAG, "onActivityCreated: did data change? ");
            });
        }

    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_best_sellers_list, container, false);
    }

}

File: app/src/main/java/io/github/maboueita/architecturecomponentssample/BestSellersListFragment.kt
package io.github.maboueita.architecturecomponentssample


import androidx.lifecycle.ViewModelProviders

import android.os.Bundle

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer

import io.github.maboueita.architecturecomponentssample.data.OverviewViewModel

class BestSellersListFragment : Fragment() {

    private var overviewViewModel: OverviewViewModel? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        overviewViewModel = ViewModelProviders.of(this).get(OverviewViewModel::class.java)
        overviewViewModel!!.init()
        Log.d("YOLO", "do we get here?")
        if (overviewViewModel!!.results != null) {
            overviewViewModel!!.results!!.observe(this, Observer {
                Log.d("YOLO", "Results are back? ${it.Lists().books?.size}")
            })

        } else {
            Log.d("YOLO ", "we back but results are null.")
        }

        fun onCreateView(inflater: LayoutInflater, container: ViewGroup,
                         savedInstanceState: Bundle): View {
            // Inflate the layout for this fragment
            return inflater.inflate(R.layout.fragment_best_sellers_list, container, false)
        }
    }
}
